cpp/oneapi/dal/table/detail/sparse_access_iface.hpp

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/array.hpp"
+
+namespace oneapi::dal::detail {
+namespace v1 {
+
+enum class csr_indexing { zero_based, one_based };
```

Comment:
`csr_indexing` and `sparse_block` are user-facing objects and should not be declared here. Move them closer to table/accessor

Reply:
It was moved to separate file